### PR TITLE
feat: check deploy tx status

### PR DESCRIPTION
**Pull request by hal3e**
close: https://github.com/FuelLabs/fuels-rs/issues/1141

This PR adds a check of the tx status when deploying contracts.
I have refactored the code a bit and added a new error variant `SqueezedOutTransactionError`. I am not sure if this should be an error. What do you think @FuelLabs/sdk-rust ?

### Checklist
- [x] I have linked to any relevant issues.
- [x] I have added necessary labels.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Review by digorithm — APPROVED** [review]
Yeah, I believe this new error should be bubbled up to the SDK user.

**Review by Br1ght0ne — DISMISSED** [review]

File: packages/fuels-core/src/types/errors.rs [hunk 1]
```diff
@@ -35,6 +35,8 @@ pub enum Error {
         revert_id: u64,
         receipts: Vec<Receipt>,
     },
+    #[error("Transaction was squeezed out")]
+    SqueezedOutTransactionError,
```
Comment (MujkicA) [inline packages/fuels-core/src/types/errors.rs:39]: Can we propagate the reason for why the tx was squeezed out?

Comment (hal3e) [reply packages/fuels-core/src/types/errors.rs:39]: great catch!

**Review by digorithm — APPROVED** [review]
:shipit:

**Review by iqdecay — APPROVED** [review]
I think it's ok to make it an SDK error, it is the same class of mistake as a revert for instance.

**Review by Salka1988 — APPROVED** [review]

**Review by MujkicA — APPROVED** [review]

**Pull request merged**